From 0173fa6c2d721b174f4ac44c2da1e8b811da1c99 Mon Sep 17 00:00:00 2001
From: Christian Jann <christian.jann@ymail.com>
Date: Fri, 13 Dec 2013 15:44:13 +0100
Subject: [PATCH 2/4] ARM: imx23-olinuxino: Add i2c support

mx23-olinuxino board has i2c interface exposed on UEXT connector. This
patch adds i2c support. Test is done with module MOD-IO2 attached on
UEXT
connector.

Signed-off-by: Fadil Berisha <f.koliqi@gmail.com>
Signed-off-by: Fabio Estevam <fabio.estevam@freescale.com>
---
V2:
Reworked as per Shawn's request. I2C instance has 3 pinmux settings
V3:
Fixed label "i2c"
V4:
Removed custom property "i2c-dma-channel" and DMA interrupt "26"
---
 arch/arm/boot/dts/imx23-olinuxino.dts |  6 ++++++
 arch/arm/boot/dts/imx23.dtsi          | 40 ++++++++++++++++++++++++++++++++++-
 2 files changed, 45 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx23-olinuxino.dts b/arch/arm/boot/dts/imx23-olinuxino.dts
index 182fa74..c66cc4f 100644
--- a/arch/arm/boot/dts/imx23-olinuxino.dts
+++ b/arch/arm/boot/dts/imx23-olinuxino.dts
@@ -73,6 +73,12 @@
 				status = "okay";
 			};
 
+			i2c: i2c@80058000 {
+				pinctrl-names = "default";
+				pinctrl-0 = <&i2c_pins_b>;
+				status = "okay";
+			};
+
 			duart: serial@80070000 {
 				pinctrl-names = "default";
 				pinctrl-0 = <&duart_pins_a>;
diff --git a/arch/arm/boot/dts/imx23.dtsi b/arch/arm/boot/dts/imx23.dtsi
index 8136057..8571993 100644
--- a/arch/arm/boot/dts/imx23.dtsi
+++ b/arch/arm/boot/dts/imx23.dtsi
@@ -317,6 +317,39 @@
 					fsl,voltage = <1>;
 					fsl,pull-up = <1>;
 				};
+
+				i2c_pins_a: i2c0@0 {
+					reg = <0>;
+					fsl,pinmux-ids = <
+						0x01e0 /* MX23_PAD_I2C_SCL__I2C_SCL */
+						0x01f0 /* MX23_PAD_I2C_SDA__I2C_SDA */
+					>;
+					fsl,drive-strength = <1>;
+					fsl,voltage = <1>;
+					fsl,pull-up = <1>;
+				};
+
+				i2c_pins_b: i2c1@1 {
+					reg = <1>;
+					fsl,pinmux-ids = <
+						0x1171 /* MX23_PAD_LCD_ENABLE__I2C_SCL */
+						0x1181 /* MX23_PAD_LCD_HSYNC__I2C_SDA  */
+					>;
+					fsl,drive-strength = <1>;
+					fsl,voltage = <1>;
+					fsl,pull-up = <1>;
+				};
+
+				i2c_pins_c: i2c2@2 {
+					reg = <2>;
+					fsl,pinmux-ids = <
+						0x2031 /* MX23_PAD_SSP1_DATA1__I2C_SCL  */
+						0x2041 /* MX23_PAD_SSP1_DATA2__I2C_SDA  */
+					>;
+					fsl,drive-strength = <1>;
+					fsl,voltage = <1>;
+					fsl,pull-up = <1>;
+				};
 			};
 
 			digctl@8001c000 {
@@ -450,8 +483,13 @@
 				status = "disabled";
 			};
 
-			i2c@80058000 {
+			i2c: i2c@80058000 {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				compatible = "fsl,imx28-i2c";
 				reg = <0x80058000 0x2000>;
+				interrupts = <27>;
+				clock-frequency = <100000>;
 				dmas = <&dma_apbx 3>;
 				dma-names = "rx-tx";
 				status = "disabled";
-- 
1.8.3.1

